### PR TITLE
Implementation of spot checks

### PR DESCRIPTION
**Pull request by novotnyjakub**
This PR will stay open until the spot checks epic is ready to be merged to `master`.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 42544d0e152cb03601422431df70c9a99271e717
  * Your subject line is longer than 50 characters

 * Commit: 41085a87021437eddc8fb684e227013051ffe68d
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/ministryofjustice/fr-staffapp

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 42544d0e152cb03601422431df70c9a99271e717
  * Your subject line is longer than 50 characters

 * Commit: 41085a87021437eddc8fb684e227013051ffe68d
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/ministryofjustice/fr-staffapp

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 42544d0e152cb03601422431df70c9a99271e717
  * Your subject line is longer than 50 characters

 * Commit: 41085a87021437eddc8fb684e227013051ffe68d
  * Your subject line is longer than 50 characters

 * Commit: 7ca5e585e6c3351aac53ac60797203de3ff87613
  * Your commit message body contains a line that is longer than 72 characters

 * Commit: 15d017499bfc448ddc79b90f79e4176102931459
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ministryofjustice/fr-staffapp

---
This message was auto-generated by https://gitcop.com

File: spec/factories/references.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FactoryGirl.define do
   factory :reference do
     application nil
+    sequence(:reference) { |n| "AB001-#{Time.zone.now.strftime('%y')}-#{n}" }
   end
 end
```
Comment (colinbruce) [inline spec/factories/references.rb]: Do we still need this factory?  
The reference model was deleted and the application factory has the sequence code now.

Comment (novotnyjakub) [inline spec/factories/references.rb]: Good spot! No, we don't and I'll remove it.

**Comment by dotemacs** [comment]
Apart from the Git style violation, looks fine

**Comment by novotnyjakub** [comment]
So the Git style validations, it doesn't allow me to have more text in the commit description. The first lines are short enough, but then it complains for the longer description, which I guess shouldn't be limited.

**Comment by novotnyjakub** [comment]
OK, I get it, it's the wrapping... I'll try to be more diligent next time, sorry.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 42544d0e152cb03601422431df70c9a99271e717
  * Your subject line is longer than 50 characters

 * Commit: 41085a87021437eddc8fb684e227013051ffe68d
  * Your subject line is longer than 50 characters

 * Commit: 7ca5e585e6c3351aac53ac60797203de3ff87613
  * Your commit message body contains a line that is longer than 72 characters

 * Commit: 15d017499bfc448ddc79b90f79e4176102931459
  * Your commit message body contains a line that is longer than 72 characters

 * Commit: 70e36e672c706b26ed82d2d0918fb2bfc371dfea
  * Your commit message body contains a line that is longer than 72 characters

 * Commit: 1cbe520c709635891e286a2ea129a58c464d6465
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ministryofjustice/fr-staffapp

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 42544d0e152cb03601422431df70c9a99271e717
  * Your subject line is longer than 50 characters

 * Commit: 41085a87021437eddc8fb684e227013051ffe68d
  * Your subject line is longer than 50 characters

 * Commit: 7ca5e585e6c3351aac53ac60797203de3ff87613
  * Your commit message body contains a line that is longer than 72 characters

 * Commit: 15d017499bfc448ddc79b90f79e4176102931459
  * Your commit message body contains a line that is longer than 72 characters

 * Commit: 70e36e672c706b26ed82d2d0918fb2bfc371dfea
  * Your commit message body contains a line that is longer than 72 characters

 * Commit: 1cbe520c709635891e286a2ea129a58c464d6465
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ministryofjustice/fr-staffapp

---
This message was auto-generated by https://gitcop.com

**Pull request merged**